File: engine/engine-model/src/main/scala/tools/packet/ui/GMEffect.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GMEffect(private var _type: Int, private var _mode: Byte) extends PacketInput {
  def theType: Int = _type

  def mode: Byte = _mode

  override def opcode(): SendOpcode = SendOpcode.ADMIN_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/ui/GMEffect.java
package tools.packet.ui;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GMEffect(Integer theType, Byte mode) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ADMIN_RESULT;
   }
}